griffon/services/core_queries.py

```diff
@@ -197,31 +197,35 @@ def execute(self, status=None) -> dict:
 def async_retrieve_sources(self, purl):
     params = {
         "limit": 300,
-        "latest_components": "True",
         "root_components": "True",
+        "released_components": "True",
         "provides": purl,
         "include_fields": "type,nvr,purl,name,namespace,download_url,related_url",
     }
-    # TODO: remove max_result
+    # TODO: remove cnt and max_result and cnt > 10000 on next stage-> prod deployment
+    cnt = self.components.count(**params)
+    if cnt > 10000:
+        return list(self.components.retrieve_list_iterator_async(max_results=5000, **params))
     return list(self.components.retrieve_list_iterator_async(max_results=10000, **params))
```

Comment:
That is the reasoning behind this? :thinking: I mean, when `cnt` is 10000, we can safely retrieve all of them and keep the 10000 max results, but when the `cnt` is ˚10001, we constraint that to 5000? Why not constraint to 10000 everything without a difference?